### PR TITLE
Allow print when it is behind the `kDebugMode` flag

### PR DESCRIPTION
**Pull request by bwilkerson**
Per our conversation.

### CODE REVIEW

File: lib/src/rules/avoid_print.dart [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import 'package:analyzer/dart/ast/ast.dart';
 import 'package:analyzer/dart/ast/visitor.dart';
 import 'package:analyzer/dart/element/element.dart';
+import 'package:linter/src/util/flutter_utils.dart';
```
Comment (pq) [inline lib/src/rules/avoid_print.dart:8]: I think this needs to get relativized...

File: test_data/rules/avoid_print.dart [hunk 2]
```diff
@@ -4,10 +4,13 @@
 
 // test w/ `dart test -N avoid_print`
 
+import 'package:flutter/foundation.dart';
+
 void main() {
   print('ha'); // LINT
   [1,2,3].forEach(print); // LINT
   Future.value('hello').then(print); // LINT
+  if (kDebugMode) print(''); // OK
```
Comment (pq) [inline test_data/rules/avoid_print.dart:13]: Maybe a case w/ brace too?

```dart
if (kDebugMode) {
  print(''); // OK
}
```

Comment (pq) [reply test_data/rules/avoid_print.dart:13]: I'm guessing we don't care about cases like:

```dart
if (!kDebugMode) {
  /// ... 
} else {
  print(''); // OK
}
```

(To be clear, I don't think we should...)

**Review by pq — APPROVED** [review]

File: lib/src/rules/avoid_print.dart (hunk 1 above)
Comment (bwilkerson) [reply lib/src/rules/avoid_print.dart]: Done

File: test_data/rules/avoid_print.dart (hunk 2 above)
Comment (bwilkerson) [reply test_data/rules/avoid_print.dart:13]: Done. And no, I wasn't planning on supporting any kind of flow control logic unless users really need it.

**Pull request merged**